arg-parsing-test.bats

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env bats
+
+source lib/assert.bash
+source deploy.sh --source-only
+
+setup() {
+	run mktemp -dt deploy_test.XXXX
+	assert_success
+	tmp=$output
+	pushd "$tmp" >/dev/null
+}
+
+teardown() {
+	popd >/dev/null
+	rm -rf "$tmp"
+}
+
+set_env_vars() {
+	# Set environment variables.
+	export GIT_DEPLOY_USERNAME=env-username
+	export GIT_DEPLOY_EMAIL=env-email
```

Comment:
I have it in my head to set those things using `git config` info, but I haven't gotten to that yet.

I was using them to help me figure out why my tests for `append_hash` were failing as well, and neglected to remove them when I got things working. No need to keep them in here, I guess. I'll just re-add them if/when I need them later.